### ISSUE TITLE
Handling implicit returns

### ISSUE THREAD

**Issue by GlowingScrewdriver** [chsasank/llama.lisp#31]
One of LLVM's conditions for a well-formed program is that every basic block must be terminated. That means that the last instruction must be a terminating instruction, like a return, branch, or anything that transfers control away from the basic block.

There is one edge case that we need to look into: the end of a function. Currently, neither Brilisp nor C-Lisp handle this

If a program is written in such a way that there is no terminating instruction at the end of a function, neither Brilisp nor C-Lisp correct it, and it is eventually Clang/LLVM that throws the error. For example, trying to compile the following program:
```scheme
(c-lisp
    (define ((fn int))
    (if #t
        (ret 0)
        (ret 1))))
```
will cause this (rather cryptic) error to be thrown by LLVM:
```
/tmp/tmp.5DCpGClmyt.ll:18:1: error: expected instruction opcode
}
^
1 error generated.
```

**Comment by GlowingScrewdriver** [comment]
I can think of a policy to handle this. Perhaps we can implement this in C-Lisp, and leave Brilisp unchanged.
* Once a function body is ready, scan it and remove labels that will produce empty basic blocks
* Insert a return statement at the end of the function if it returns void
* Detect and report the problem if the function doesn't return void

**Comment by chsasank** [comment]
Let's just not allow implicit returns in c-lisp.

**Comment by GlowingScrewdriver** [comment]
Alright. But we still have to deal with things like
```scheme
(define ((main int))
    (if #t
       (ret 1)
       (ret 0)))
```
Because if we don't deal with it, we are encouraging code like this:
```scheme
(define ((main int))
    (if #t
       (ret 1))
    (ret 0))
```
Either we can scan the function body and remove empty labels, or we can implement an `unreachable` instruction like LLVM.

**Comment by GlowingScrewdriver** [comment]
LLVM's `unreachable` instruction marks a location in the code as unreachable; but it is also a terminating instruction. So, if we implemented `unreachable`, we could expect the programmer to be explicit and write code like this:
```scheme
(define ((main int))
    (if #t
       (ret 1)
       (ret 0))
       (unreachable))
```

**Comment by chsasank** [comment]
Write equivalent C code, try compiling and put llvm IR output if it does

**Comment by chsasank** [comment]
Closed by #49
